### ISSUE TITLE
update needed for Inkscape version 1.0

### ISSUE THREAD

**Issue by brynnd** [fablabnbg/inkscape-silhouette#90]
Hello,
Thank you for providing this extension for Inkscape users!

This is just to let you know that most 3rd party Inkscape extensions, like this one, probably will not work with the upcoming new Inkscape version, the long-awaited version 1.0.

Here is the info you need to update this extension, so that it will work with 1.0 and future versions.  
https://wiki.inkscape.org/wiki/index.php?title=Updating_your_Extension_for_1.0

If you have further questions, you can contact Inkscape developers via [mailing lists](https://lists.inkscape.org/postorius/lists/?all-lists), [forum](https://inkscape.org/forums/extensions/), or the [chatroom](https://chat.inkscape.org/channel/inkscape_extensions).

If you have already updated it, please disregard this message.

All best,
brynn

**Comment by jnweiger** [comment]
Thanks for the reminder. We discussed that already on the last inkscape hackfests. I simply lack time to do it. Patches welcome.

**Comment by t0b3** [comment]
@jnweiger PR #94 helps to solve this issue

**Comment by iromero91** [comment]
For what it's worth, current master works perfectly on version 1.0 (tested on debian testing). Only issue is a huge amount of deprecation warnings when running the script.

**Comment by t0b3** [comment]
just wondering what happens if you try this?
`sendto_silhouette.py --dummy=True examples/testcut_square_triangle.svg`
- any errors?

**Comment by t0b3** [comment]
> For what it's worth, current master works perfectly on version 1.0 (tested on debian testing). Only issue is a huge amount of deprecation warnings when running the script.

@iromero91 just wondering what happens if you try this?
`sendto_silhouette.py --dummy=True examples/testcut_square_triangle.svg`
- any errors?

**Comment by iromero91** [comment]
@t0b3 I get this when running from the console.
```
$ python sendto_silhouette.py --dummy=True examples/testcut_square_triangle.svg
sendto_silhouette.py:126: DeprecationWarning: inkex.localize was moved to inkex.localization.localize
  inkex.localize()	# 0.91
sendto_silhouette.py:270: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--active-tab', action = 'store', dest = 'active_tab',
sendto_silhouette.py:272: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-d', '--dashes',
sendto_silhouette.py:275: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-a', '--autocrop',
sendto_silhouette.py:278: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-b', '--bbox', '--bbox-only', '--bbox_only',
sendto_silhouette.py:281: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-c', '--bladediameter',
sendto_silhouette.py:284: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-C', '--cuttingmat', action = 'store',
sendto_silhouette.py:287: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-D', '--depth',
sendto_silhouette.py:290: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--dummy',
sendto_silhouette.py:293: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-g', '--strategy',
sendto_silhouette.py:297: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-l', '--sw_clipping',
sendto_silhouette.py:300: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-m', '--media', '--media-id', '--media_id',
sendto_silhouette.py:306: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-o', '--overcut',
sendto_silhouette.py:309: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-M', '--multipass',
sendto_silhouette.py:312: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-p', '--pressure',
sendto_silhouette.py:315: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-P', '--sharpencorners',
sendto_silhouette.py:318: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--sharpencorners_start',
sendto_silhouette.py:321: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--sharpencorners_end',
sendto_silhouette.py:324: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-r', '--reversetoggle',
sendto_silhouette.py:327: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-s', '--speed',
sendto_silhouette.py:330: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option( "-S", "--smoothness", action="store", type="float",
sendto_silhouette.py:332: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-t', '--tool', action = 'store',
sendto_silhouette.py:334: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-T', '--toolholder', action = 'store',
sendto_silhouette.py:336: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-V', '--version',
sendto_silhouette.py:339: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-w', '--wait', '--wait-done', '--wait_done',
sendto_silhouette.py:342: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-x', '--x-off', '--x_off', action = 'store',
sendto_silhouette.py:344: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-y', '--y-off', '--y_off', action = 'store',
sendto_silhouette.py:346: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-R', '--regmark',
sendto_silhouette.py:349: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--regsearch',
sendto_silhouette.py:352: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-X', '--reg-x', '--regwidth', action = 'store',
sendto_silhouette.py:354: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-Y', '--reg-y', '--reglength', action = 'store',
sendto_silhouette.py:356: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--rego-x',  '--regoriginx',action = 'store',
sendto_silhouette.py:358: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('--rego-y', '--regoriginy', action = 'store',
sendto_silhouette.py:360: DeprecationWarning: Effect.OptionParser or `optparse` has been deprecated and replaced with `argparser`.You must change `self.OptionParser.add_option` to `self.arg_parser.add_argument`; the arguments are similar.
  self.OptionParser.add_option('-e', '--endposition', '--end-postition',
sendto_silhouette.py:1238: DeprecationWarning: Effect.affect is now `Effect.run()`. The `output` argument has changed.
  e.affect()
status=none
device version: 'None'
7 self.docHeight= 1133.8582677165355
8 self.docWidth= 1511.8110236220473
sendto_silhouette.py:517: DeprecationWarning: simpletransform.composeParents -> elem.composed_transform() or elem.transform * Transform(mat)
  transform = composeParents(node, IDENTITY_TRANSFORM)
sendto_silhouette.py:518: DeprecationWarning: simpletransform.composeTransform -> Transform(M1) * Transform(M2)
  transform = composeTransform(self.docTransform, transform)
sendto_silhouette.py:519: DeprecationWarning: simpletransform.composeTransform -> Transform(M1) * Transform(M2)
  transform = composeTransform(extra_transform, transform)
sendto_silhouette.py:615: DeprecationWarning: inkex.etree was removed, use "from lxml import etree"
  newpath = inkex.etree.Element( inkex.addNS( 'path', 'svg' ) )
sendto_silhouette.py:632: DeprecationWarning: simplepath.formatPath -> str(element.path) or str(Path(array))
  newpath.set( 'd', simplepath.formatPath( a ) )
Traceback (most recent call last):
  File "sendto_silhouette.py", line 1238, in <module>
    e.affect()
  File "/usr/share/inkscape/extensions/inkex/deprecated.py", line 180, in affect
    return self.run(args=args)
  File "/usr/share/inkscape/extensions/inkex/base.py", line 123, in run
    self.save_raw(self.effect())
  File "sendto_silhouette.py", line 1034, in effect
    self.recursivelyTraverseSvg( self.document.getroot() )
  File "sendto_silhouette.py", line 533, in recursivelyTraverseSvg
    self.recursivelyTraverseSvg( node, parent_visibility=v )
  File "sendto_silhouette.py", line 533, in recursivelyTraverseSvg
    self.recursivelyTraverseSvg( node, parent_visibility=v )
  File "sendto_silhouette.py", line 632, in recursivelyTraverseSvg
    newpath.set( 'd', simplepath.formatPath( a ) )
  File "/usr/share/inkscape/extensions/inkex/deprecated.py", line 257, in _inner
    return func(*args, **kwargs)
  File "/usr/share/inkscape/extensions/inkex/deprecated-simple/simplepath.py", line 36, in formatPath
    return str(Path(a))
  File "/usr/share/inkscape/extensions/inkex/paths.py", line 1034, in __init__
    self.append(PathCommand.letter_to_class(item[0])(*item[1]))
  File "/usr/share/inkscape/extensions/inkex/paths.py", line 109, in letter_to_class
    return PathCommand._letter_to_class[letter]
KeyError: 'M '
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/tty' mode='w' encoding='UTF-8'>
```

**Comment by bianchifan** [comment]
While runnin' with some probs on Ubuntu 19.10 / Inkscape 0.92.5 it does not on Ubuntu 20.04 Inkscape 1.0.
I must update Inkscape to 1.0 because 0.92.5 refused every extension.
1.0 needs updated extensions, several 3rd party ones do not run - inkscape-silhouette belongs to them..

![Bildschirmfoto von 2020-05-15 17-59-36](https://user-images.githubusercontent.com/17491743/82121275-fb43e300-978b-11ea-8d5f-ed84a4b7dc5d.png)

EDIT:

Another problem with Inkscape 1.0 : setup.py
When installed from deb setup.py is registered and prohibits an update from inkscape 0.92 to 1.0 for inkscape has a setup.py also.
So the setup.py should be named setup-silhouette.py or setup-inkscape-silhouette.py.

**Comment by t0b3** [comment]
@bianchifan see #100 for issue with `setup.py`. 
try current master with inkscape wich runs extensions with python3. #94 is partly solving this issue, though more work is needed to be fully inkscape 1.0 compatible.

**Comment by bianchifan** [comment]
> @bianchifan see #100 for issue with setup.py.

I wrote it for information only, I found a workaround for my own.

> try current master with inkscape wich runs extensions with python3

That''s what I did in the meantime.. ;)
I had to delete ma current "silhouette" folder completely and took the actual master's.

Afterwards the plugin runs but produces only bullshit.
I tried two files with a bunch of layers but only one visible.. -> shit!
I took one file and deleted all layers but the cutting one.. -> shit!

I tried the same files on another partition with Ubuntu Studio 19.10(SAME! HOME partition!) -> success!
Please have a view at the pic below, the upper one is inkscape 1.0, the lower one is 0.92.5.

![DSC03404](https://user-images.githubusercontent.com/17491743/82123987-a78dc580-979c-11ea-8b49-8525683e50b4.JPG)

Paper is DIN A4 landscape, the image should be 15,8 cm wide.
Both plots done with the reduced cutting-layer-only-file.
[CooloH.zip](https://github.com/fablabnbg/inkscape-silhouette/files/4638665/CooloH.zip)

**Comment by t0b3** [comment]
@iromero91 thank you for your feedback. 
you [reproduce](https://github.com/fablabnbg/inkscape-silhouette/issues/90#issuecomment-628982616) the `key error 'M '` I detected using travis.

@jnweiger see #99 which will help a lot to detect and fix this kind of error and warnings while keeping compatiblity with python 2 and inkscape 0.92.x

**Comment by t0b3** [comment]
> Please have a view at the pic below, the upper one is inkscape 1.0, the lower one is 0.92.5.
> 
> ![DSC03404](https://user-images.githubusercontent.com/17491743/82123987-a78dc580-979c-11ea-8b49-8525683e50b4.JPG)
> 
> Paper is DIN A4 landscape, the image should be 15,8 cm wide.
> Both plots done with the reduced cutting-layer-only-file.

Hmm, this seems really odd. To catch the failing point it might be interesting to compare the output from both environments calling `sendto_silhouette.py --dummy=True <file.svg>` which will give you a dump file at `/tmp/silhouette.dump` Is the content of both dump files identical?

**Comment by bianchifan** [comment]
Your wish let me work...but keep in mind: the extension is exactly the same, both Version use the SAME!! HOME partition.

Ubuntu Stdio 19.10 with inkscape 0.92.5 produces a nice output:
[silhouette.dump.zip](https://github.com/fablabnbg/inkscape-silhouette/files/4639163/silhouette.dump.zip)

On Ubuntu 20.04 / inkscape 1.0 python wont be found this way..
![Bildschirmfoto von 2020-05-16 22-26-16](https://user-images.githubusercontent.com/17491743/82129604-585b8b00-97c4-11ea-9c91-50a57a9868fd.png)

Called explicitly with python3 did run but output is different...
[terminal-output-20.04.zip](https://github.com/fablabnbg/inkscape-silhouette/files/4639202/terminal-output-20.04.zip)

and the dump
[silhouette.dump-20.04.zip](https://github.com/fablabnbg/inkscape-silhouette/files/4639191/silhouette.dump-20.04.zip)

**Comment by EtherGraf** [comment]
Following the wrong scaling there seems to be a clipping with the correct vertical dimensions.
It is known, that all lines which cross the clipping border are dismissed complete length.

**Comment by t0b3** [comment]
@brynnd @iromero91 resolved with #118. You re welcome to try the latest code and provide more feedback.

**Comment by bianchifan** [comment]
I see, issue is closed but I needed some time for my test of the actual version (due to system probs and installing Ubuntu 20.10 from scratch)

Scenario: Ubuntu 20.20 fresh install, inkscape 1.0.1, Silhouette plugin from 2021.01.10

1st. Scaling seems correct now.
2nd. Positioning fails.
3rd. Edging failed.
4th. Registration marks -> abort.

I tried 2 svgs, the "Fox" decals from the scaling issue and a new one for a friends project, Audi Quattro S1 1:10.
Last December I realized the Quattro project with inkcut. Was a little complicated for inkcut doen't support print&cut.

Project file
[Audi Quattro S1 - Decals.zip](https://github.com/fablabnbg/inkscape-silhouette/files/5826465/Audi.Quattro.S1.-.Decals.zip)

AS I wrote, scaling is ok now!
Positioning fails, objects are shifted directions left and top. I tried to solve it with an offset correction but I wasn't successful.
I do not understand who offset is working..., the results always differ to my expectation.
Another problem: Some rounded corners (rectangles??) were plotted sharp.

![DSC03651](https://user-images.githubusercontent.com/17491743/104852078-ee6a2700-58f8-11eb-8a0e-4a5ca84f6fbd.jpg)

Trying to cut with regmarks let the cut stop immediately, no regmarks found.
But the cutter never tried to search....
At least a deprecation warning came up.

![Bildschirmfoto von 2021-01-17 16-18-38](https://user-images.githubusercontent.com/17491743/104852188-b0213780-58f9-11eb-94fd-d97e77091f7e.png)

**Comment by t0b3** [comment]
@bianchifan it seems you found multiple issues:
- [ ] Positioning / Edging / Registration marks abort
   - I guess these are all related, right?
   - what plotter do you use?
   - who can reproduce this?
- [ ] Rounded corners: cut sharp
   - not a new issue, right? [any contributions to improve this](https://github.com/fablabnbg/inkscape-silhouette/blob/master/sendto_silhouette.py#L637-L679) are most welcome...
- [x] Deprecation warning: apport/report.py:13
  - out of scope for this plugin. needs to be solved upstream.

**Comment by bianchifan** [comment]
>       * what plotter do you use?
Silhouette CAMEO 2

>     * [ ]  Rounded corners: cut sharp
>       
>       * not a new issue, right? [any contributions to improve this](https://github.com/fablabnbg/inkscape-silhouette/blob/master/sendto_silhouette.py#L637-L679) are most welcome...
I just checked the svg, the "sharp" cutted object are rectangles, indeed.
If this is an old issue, I can solve it by an exchange with paths :;, maybe I didn't use rects with inkscape 0.92,, I do not remember.

>     * [x]  Deprecation warning: apport/report.py:13
>       
>       * out of scope for this plugin. needs to be solved upstream.
Print&Cut did never work for me, I posted an issue for that some time ago.
BUT the cutter tried to find the marks.
Sometimes they were not found after a lot of noise, sometimes they were found but the result was crap.
Now, with the actual plugin + inkscape 1.0.1/Ubuntu 20.10 it doesnot start to search, the message "..marks not found.." appears immediately on the cutter screen.
